src/container/srv_target.c

```diff
@@ -1452,14 +1453,16 @@ cont_child_aggregate(struct ds_cont_child *cont, uint64_t *sleep)
 	 * sc_aggregation_min != DAOS_EPOCH_MAX means we need to aggregate
 	 * from epoch 0 because some snapshot was deleted.
 	 */
-	if (cont->sc_aggregation_min != DAOS_EPOCH_MAX) {
+	if (cont->sc_aggregation_min != DAOS_EPOCH_MAX ||
+	    cont->sc_pool->spc_reset_aggregation) {
 		epoch_min = 0;
 		/*
 		 * Set sc_aggregation_min to non-zero so that we can tell
 		 * if another snapshot deletion happened when this round of
 		 * aggregation done.
 		 */
 		cont->sc_aggregation_min = hlc;
```

Comment:
The 'sc_aggregation_min' will be mistakenly set to 'hlc' when 'spc_reset_aggregation' == 1, and it won't be set back to DAOS_EPOCH_MAX if cont_child_aggregate() returns before reaching the function ending.